src/django_components/apps.py

```diff
@@ -49,3 +47,19 @@ def ready(self) -> None:
 
         # Register the dynamic component under the name as given in settings
         registry.register(app_settings.DYNAMIC_COMPONENT_NAME, DynamicComponent)
+
+
+# See https://github.com/EmilStenstrom/django-components/issues/586#issue-2472678136
+def _watch_component_files_for_autoreload() -> None:
+    from django_components.util.loader import get_component_dirs
+
+    component_dirs = set(get_component_dirs())
+
+    def template_changed(sender: Any, file_path: Path, **kwargs: Any) -> None:
```

Comment:
This function was moved from `util/misc.py`. Also, the logic itself changed. Before, I tried to use the `autoreload_started` signal, and `sender.extra_files.add()` to watch the files.

Now, I connect to the `file_changed` signal, and use `trigger_reload()`.

I think I've seen this in either Django or Tetra codebase.